include/seqan3/io/alignment_file/misc.hpp

```diff
@@ -21,4 +21,68 @@ namespace seqan3
+ *   has neither the seqan3::sam_flag::secondary_alignment nor the seqan3::sam_flag::supplementary_alignment flag value
+ *   set (satisfies `FLAG & 0x900 == 0 `). This line is called the **primary alignment** of the read.
+ * * seqan3::sam_flag::secondary_alignment (bit `0x100`) marks the alignment not to be used in certain analyses when
+ *   the tools in use are aware of this bit. It is typically used to flag alternative mappings when multiple mappings
+ *   are presented in a SAM.
+ * * seqan3::sam_flag::supplementary_alignment (bit `0x800`) indicates that the corresponding alignment line is part
+ *   of a chimeric alignment. If the SAM/BAM file corresponds to long reads (nanopore/pacbio) this happens when
+ *   reads are split before being aligned and the best matching part is marked as primary, while all other aligned
+ *   parts are marked supplementary.
+ * * seqan3::sam_flag::unmapped (bit `0x4`) is the only reliable place to tell whether the read is unmapped.
+ *   If seqan3::sam_flag::unmapped is set, no assumptions can be made about RNAME, POS, CIGAR, MAPQ, and
+ *   seqan3::sam_flag::proper_pair, seqan3::sam_flag::secondary_alignment, and seqan3::sam_flag::supplementary_alignment
+ *   (bits `0x2`, `0x100`, and `0x800`).
+ * * seqan3::sam_flag::on_reverse_strand (bit `0x10`) indicates  whether the read sequence has been reverse complemented
+ *   and the quality string is reversed.  When bit seqan3::sam_flag::unmapped (`0x4`) is unset, this
+ *   corresponds to the strand to which the segment has been mapped: seqan3::sam_flag::on_reverse_strand (bit `0x10`)
+ *   unset indicates the forward strand, while set indicates the reverse strand. When seqan3::sam_flag::unmapped (`0x4`)
+ *   is set, this indicates whether the unmapped read is stored in its original orientation as it came off the
+ *   sequencing machine.
+ * * seqan3::sam_flag::first_in_pair and seqan3::sam_flag::second_in_pair (bits `0x40` and `0x80`) reflect the read
+ *   ordering within each template inherent in the sequencing technology used. If seqan3::sam_flag::first_in_pair and
+ *   seqan3::sam_flag::second_in_pair (`0x40` and `0x80`) are both set, the read is part of a linear template, but it
+ *   is neither the first nor the last read. If both are unset, the index of the read in the template is unknown.
+ *   This may happen for a non-linear template or when this information is lost during data processing.
+ * * If seqan3::sam_flag::paired (bit `0x1`) is unset, no assumptions can be made about seqan3::sam_flag::proper_pair,
+ *   seqan3::sam_flag::mate_unmapped, seqan3::sam_flag::mate_on_reverse_strand, seqan3::sam_flag::first_in_pair and
+ *   seqan3::sam_flag::second_in_pair (bits `0x2`, `0x8`, `0x20`, `0x40` and `0x80`).
+ *
+ * \sa https://broadinstitute.github.io/picard/explain-flags.html
+ */
+enum class sam_flag : uint16_t
+{
+   paired                  = 0x1,   //!< The aligned read is paired (paired-end sequencing).
```

Comment:
Maybe add `none` with value zero so it can be initialised and reset without passing `{0}` which is not very expressive for an enum.